Signal/src/ViewControllers/SelectThreadViewController.m

```diff
@@ -216,33 +195,31 @@ - (void)updateTableContents
     self.tableViewController.contents = contents;
 }
 
-- (void)contactWasSelected:(Contact *)contact
+- (void)signalAccountWasSelected:(SignalAccount *)signalAccount
 {
-    OWSAssert(contact);
+    OWSAssert(signalAccount);
     OWSAssert(self.delegate);
 
-    // TODO: Use ContactAccount.
-    NSString *recipientId = contact.textSecureIdentifiers.firstObject;
-    
-    if ([self isRecipientIdBlocked:recipientId] &&
-        ![self.delegate canSelectBlockedContact]) {
-        
+    ContactsViewHelper *helper = self.contactsViewHelper;
+
+    if ([helper isRecipientIdBlocked:signalAccount.recipientId] && ![self.delegate canSelectBlockedContact]) {
+
         __weak SelectThreadViewController *weakSelf = self;
-        [BlockListUIUtils showUnblockContactActionSheet:contact
-                                     fromViewController:self
-                                        blockingManager:self.blockingManager
-                                        contactsManager:self.contactsManager
-                                        completionBlock:^(BOOL isBlocked) {
-                                            if (!isBlocked) {
-                                                [weakSelf contactWasSelected:contact];
-                                            }
-                                        }];
+        [BlockListUIUtils showUnblockSignalAccountActionSheet:signalAccount
+                                           fromViewController:self
+                                              blockingManager:helper.blockingManager
+                                              contactsManager:helper.contactsManager
+                                              completionBlock:^(BOOL isBlocked) {
+                                                  if (!isBlocked) {
+                                                      [weakSelf signalAccountWasSelected:signalAccount];
+                                                  }
+                                              }];
         return;
     }
 
     __block TSThread *thread = nil;
     [[TSStorageManager sharedManager].dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
-        thread = [TSContactThread getOrCreateThreadWithContactId:recipientId transaction:transaction];
+        thread = [TSContactThread getOrCreateThreadWithContactId:signalAccount.recipientId transaction:transaction];
```

Comment:
just a rant: 

I'd like to move towards passing SignalAccounts (or something 1:1 with it) in pretty much all the places where we're passing e164 strings around.

e.g. to avoid the "how do I know if this is e164 or not?" problem that you have when you have a parameter named "phoneNumber"

and for future work where we can't assume a recipient id is a e164 phone number.

e.g. in this case `[TSContactThread getOrCreateThreadWithSignalAccount:signalAccount transaction:transaction];

I am *not* advocating that we make any specific changed WRT that in this PR, or all at once, as it will touch a bajillion places, but if you agree, I'd like to keep it in mind for future work.